### PR TITLE
Add description of phases to the workflow

### PR DESCRIPTION
**Pull request by bocekm**
- having the description of the phases right in the workflow definition file makes it easier for the actor developer to choose the right one without the need to go to the documentation.
- right now it's just a copy&paste from https://leapp-to.github.io/upgrades and needs to be updated to be more clear what each phase is about

### CODE REVIEW

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 1]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
                             Policies.Retry.Phase)
         flags = Flags()
 
     class ChecksPhase(Phase):
+        '''Inhibitors - Blockers, Shared Objects, Binaries, Languages,
+        Packages, Load Balance Support, User Management, Version Control
+        Systems, FreeRADIUS, IPA Bind Dyndb LDAP, SSSD, NTP, OpenLDAP, YPSERV,
+        Network, Databases, SELinux, Storage, System, HTTPD, OpenSSH, Postfix,
+        SQUID, DoveCot, Bind9 Configuration, RSYSLOG.
+        '''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: The description is vague. This is not only about inhibitors. Inhibitors are small part of actions that should happened here. Here should be put all user questions, should be checked what should be done and it should be planned. basically, it's subset of actions we usually do in PA during the check of the system. Just no actions should be done here.

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 2]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
     description = '''No description has been provided for the InplaceUpgrade workflow.'''
 
     class FactsCollectionPhase(Phase):
+        '''Operating System Version Detection, IP, Hostname, RPMs, Services,
+        Augeas.
+        '''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: Get info about the system - what is installed, settings, ... IOW, scan the system to get facts about it.

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 3]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
     class AttachPackageReposPhase(Phase):
+        '''Make sure the repositories of the new system are available.'''
         name = 'AttachPackageRepos'
         filter = TagFilter(AttachPackageReposPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class PlanningPhase(Phase):
+        '''RPM Install transaction (feasibility checks).'''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: This phase looks completely useless from the point of the current design.

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 4]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
         name = 'Planning'
         filter = TagFilter(PlanningPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class DownloadPhase(Phase):
+        '''Download of all things needed (RPMs, Kernel images, Anaconda image
+        (very big maybe), etc.)
+        '''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: here we do everything connected to the transaction - get target rpm repositories, check the rpm transaction, download all packages... 
From that point, we should rename the phase in future to fit better to what we do here.

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 5]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
         name = 'Download'
         filter = TagFilter(DownloadPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class InterimPreparationPhase(Phase):
+        '''Preparation of initial ramdisk (if required).
+        Setup bootloaderGrub.
+        '''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: What about:,,Preparation of system to be able to proceed the upgrade after reboot. E.g. prepare initial ramdisk, setup bootloader, ...).

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 6]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
         name = 'InterimPreparation'
         filter = TagFilter(InterimPreparationPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags(request_restart_after_phase=True)
 
     class InitRamStartPhase(Phase):
+        '''Reboot into the next system therefore not a real step, however the
+        before and post parts are interesting to have for hooking.
+        '''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: From the desciprtion, really not sure what about it is. Anyway, this phase is about the boot - not reboot, that's action connected to previous phase only. You are booting here.

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 7]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
     class StoragePhase(Phase):
+        '''Mount all storage points.'''
         name = 'Storage'
         filter = TagFilter(StoragePhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class LateTestsPhase(Phase):
+        '''Tests that need to be executed later.'''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: What about: last tests before the upgrade itself, that has to be done with the new kernel.

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 8]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
         name = 'LateTests'
         filter = TagFilter(LateTestsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class PreparationPhase(Phase):
+        '''YUM Configurations. Backup not upgradable configurations (No Verify
+        Configs).
+        '''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: No Verify Configs != not upgradable configuration

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 9]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
         name = 'RPMUpgrade'
         filter = TagFilter(RPMUpgradePhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class ApplicationsPhase(Phase):
+        '''Application of changes on configuration files, move various
+        directores, etc.
+        '''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: Provide all neccessary steps to finish upgrade of particular native applications. Including move/rename of configuration files (if required), modify configuration of applications to be able to run correctly (if possible, to be set up as close as possible to original behaviour), ....

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 10]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
+        upgraded.
+        '''
         name = 'ThirdPartyApplications'
         filter = TagFilter(ThirdPartyApplicationsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class FinalizationPhase(Phase):
+        '''Plan SELinux relabeling. Labels Bootloader settings.'''
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: Actually, we probably will do SELinux relabeling during this phase in future. What about: "Rest of actions that should be done before the reboot into the ugpraded system. E.g. selinux relabeling, ...."

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 11]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
+        before and post parts are interesting to have for hooking.
+        '''
         name = 'InitRamStart'
         filter = TagFilter(InitRamStartPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class NetworkPhase(Phase):
+        '''Bring up the network.'''
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: Might be unnecessary here, it's possible that we will have this done already before we start the workflow

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 12]
```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
...
         name = 'Network'
         filter = TagFilter(NetworkPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class StoragePhase(Phase):
+        '''Mount all storage points.'''
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: Might be unnecessary here, it's possible that we will have this done already before we start the workflow

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py (hunk 7 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: and new systemd

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py (hunk 8 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: I think that this should be called PreTransaction and should consider to apply workarounds etc that are needed to make the transaction successful for example.

**Comment by pirat89** [comment]
@vinzenz I keep rename of `PreparationPhase` for another PR or separate commit. For now, just kept as it is for review.

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py (hunk 12 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: commented out already.

**Comment by bocekm** [comment]
Squashed.

**Pull request merged**